shell/platform/android/io/flutter/embedding/engine/mutatorsstack/FlutterMutatorView.java

```diff
@@ -142,6 +143,12 @@ public void draw(Canvas canvas) {
       pathCopy.offset(-left, -top);
       canvas.clipPath(pathCopy);
     }
+
+    // Apply the final opacity value on the parent canvas.
+    Paint transparency = new Paint();
+    transparency.setAlpha((int) (mutatorsStack.getFinalOpacity() * 255));
+    this.setLayerType(LAYER_TYPE_HARDWARE, transparency);
```

Comment:
CC @blasten for confirming this is OK

Reply:
does it need to create a new Paint object on every `draw` call?   does this work?

In the constructor:
```java 
paint = new Paint();
setLayerType(LAYER_TYPE_HARDWARE, paint);
```

onDraw:

```java
paint.setAlpha((int) (mutatorsStack.getFinalOpacity() * 255));
```

Reply:
seems like this should work from looking at the android code, but please give it a try and confirm

Reply:
@blasten You're right, we don't need to create the paint object on every draw call. 
We do however need to set the layer type every time, moving it into the constructor does not seem to be enough.

Reply:
I have updated the code regarding this